core/domain/exp_domain.py

```diff
@@ -2616,6 +2618,64 @@ def _convert_states_v51_dict_to_v52_dict(
 
         return states_dict
 
+    @classmethod
+    def _convert_states_v52_dict_to_v53_dict(
+        cls, states_dict: Dict[str, state_domain.StateDict]
+    ) -> Dict[str, state_domain.StateDict]:
+        """Converts from version 52 to 53. Version 53 makes sure the links used
+        in the non-interactive link component are all 'https' links.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+
+        for state_dict in states_dict.values():
+            content_html_list = [state_dict['content']['html']]
+
+            for html_content in content_html_list:
+                soup = bs4.BeautifulSoup(html_content, 'html.parser')
+                links = soup.find_all('oppia-noninteractive-link')
+
+                acceptable_schemes = ['https', '']
+
+                for link in links:
+                    lnk_attr = link.get('url-with-value')
+                    txt_attr = link.get('text-with-value')
+
+                    if lnk_attr is None:
+                        # Delete the link.
+                        link.extract()
+                    if txt_attr is None:
+                        # Set link text to be the url itself.
+                        txt_attr = lnk_attr
+
+                    lnk = lnk_attr.replace('&quot;', '')
+                    txt = txt_attr.replace('&quot;', '')
+
+                    # If text is empty and the link is not.
```

Comment:
Better to remove the tag incase the `text-with-value` is empty. I was having a discussion with @seanlip regarding this. You can find it [here](https://docs.google.com/document/d/1WAkQ8P438xr7yhEz6Y1p5k3hIWyjmVZRyS3ozGxEM4g/edit?disco=AAAAb5YHOJY)

Reply:
Done